utils_linux.go

```diff
@@ -123,11 +123,13 @@ func setupIO(process *libcontainer.Process, container *libcontainer.Container, c
 			}
 			uc, ok := conn.(*net.UnixConn)
 			if !ok {
+				conn.Close()
```

Comment:
This cannot fail, and to be honest the following error should be a `panic` imho.